src/components/List/Section.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+
+import styled from "styled-components";
+
+import { spacing, colors, typography } from "../../theme";
+
+import { Row, Column } from "../Grid";
+import { SecondaryText, BoldText, Link } from "../Text";
+import { mediumAndUp } from "../../theme/mediaQueries";
+import { sectionItemShape } from "./shape";
+
+const SectionContainer = styled.div`
+  padding: ${spacing.moderate};
+  border-bottom: 1px solid ${colors.lightGray};
+
+  ${mediumAndUp`
+    border: none;
+    padding: ${spacing.moderate} 0 ${spacing.moderate} 0;
+  `};
+`;
+
+const ItemContainer = styled(Row)`
+  align-items: center;
+  cursor: ${props => (props.onClick ? "pointer" : "default")};
+  &:not(:last-of-type) {
+    margin-bottom: ${spacing.moderate};
+  }
+`;
+
+const IconContainer = styled.div`
+  padding-right: ${spacing.cozy};
+`;
+
+const TitleContainer = styled(Column)`
+  padding-bottom: ${spacing.moderate};
+`;
+
+const TitleText = styled(BoldText)`
+  font-size: ${typography.size.uno};
+  color: ${colors.onyx.light};
+  text-transform: uppercase;
+`;
+
+const Section = ({ title, items }) => (
+  <SectionContainer>
+    <TitleContainer>
+      <TitleText>{title}</TitleText>
+    </TitleContainer>
+
+    {items.map(item => (
+      <ItemContainer
+        key={item.title}
+        role="link"
+        aria-label="Section Item"
+        onClick={item.onItemClick ? item.onItemClick : null}
+      >
+        <IconContainer>{item.icon}</IconContainer>
+        <div>
+          {item.onItemClick ? (
+            <Link href="/">{item.title}</Link>
+          ) : (
+            <SecondaryText>{item.title}</SecondaryText>
+          )}
+          {item.subTitle ? (
+            <SecondaryText>{item.subTitle}</SecondaryText>
+          ) : null}
+        </div>
+      </ItemContainer>
+    ))}
```

Comment:
I think you should replace the `items` prop with `children` here to decouple the implementation